base/src/main/java/org/aya/resolve/context/ContextUnit.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020-2023 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.resolve.context;
+
+import org.aya.concrete.stmt.Stmt;
+import org.aya.ref.AnyVar;
+import org.aya.ref.DefVar;
+import org.aya.ref.LocalVar;
+import org.jetbrains.annotations.NotNull;
+
+public sealed interface ContextUnit {
```

Comment:
In this class we track where a symbol comes from. 
But it seems we don't check if a symbol is usable by judging its home?

Reply:
Okay, I see two instanceof checks of `ContextUnit.Exportable`, in `GlobalSymbol` and `ModuleLikeContext`.

Does this mean we should only track the accessibility?

Reply:
They track the accessibility and limit the type of var:
* We can only export/import `DefVar`s
* Only `DefVar`s have accessibility (I won't to see a `public GeneralizedVar`)

Reply:
Do you think we need variants other than `Exportable` and `NotExportable`? From the current code, there is only one non-trivial usage of `ContextUnit` and it is just instanceof check of `Exportable`. I guess we only need these two: one for all DefVars with accessibility, and the other for the rest cases.

Correct me if there are other design considerations.

Reply:
Then a `LocalVar` **is able to** be added to ModuleSymbol which is weird.
These variants prevent the programmer write stupid code.